clang/lib/StaticAnalyzer/Checkers/StreamChecker.cpp

```diff
@@ -1034,16 +1034,16 @@ void StreamChecker::preWrite(const FnDescription *Desc, const CallEvent &Call,
   C.addTransition(State);
 }
 
-static std::optional<QualType> getPointeeType(const MemRegion *R) {
+static QualType getPointeeType(const MemRegion *R) {
   if (!R)
-    return std::nullopt;
+    return {};
   if (const auto *ER = dyn_cast<ElementRegion>(R))
-    return ER->getElementType();
+    return ER->getElementType()->getCanonicalTypeUnqualified();
```

Comment:
Consider moving the `->getCanonicalTypeUnqualified()` calls into `escapeByStartIndexAndCount()` because that's the only place where it will be relevant. (This would eliminate the code duplication.)

Reply:
Actually, now that I don't compare the type against a concrete alternative, it's no longer relevant. I'll just drop it. Thanks for spotting.